File: backend/ws-service/src/main/java/com/example/wsservice/chat/dto/ReactionResultDto.java
package com.example.wsservice.chat.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/** chat-service 반응 토글 응답. */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReactionResultDto {
    private Long chatId;
    private String roomId;
    private String authorUserId;
    private boolean added;
    private String emoji;
    private List<ReactionSummaryDto> reactions;
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/chat/dto/ReactionResultDto.kt
package com.example.wsservice.chat.dto

/** chat-service 반응 토글 응답. */
data class ReactionResultDto(
    var chatId: Long? = null,
    var roomId: String? = null,
    var authorUserId: String? = null,
    var added: Boolean = false,
    var emoji: String? = null,
    var reactions: List<ReactionSummaryDto>? = null,
)
